src/orquestra/integrations/cirq/simulator/_base.py

```diff
@@ -0,0 +1,213 @@
+        else:
+            wavefunction = self.get_wavefunction(circuit).amplitudes
+
+            # Pyquil does not support PauliSums with no terms.
+            if len(qubit_operator.terms) == 0:
+                return ExpectationValues(np.zeros((0,)))
+
+            values = []
+
+            for pauli_term in qubit_operator:
+                sparse_pauli_term_ndarray = get_sparse_operator(
+                    pauli_term, n_qubits=circuit.n_qubits
+                ).toarray()
+                if np.size(sparse_pauli_term_ndarray) == 1:
+                    expectation_value = sparse_pauli_term_ndarray[0][0]
+                    values.append(expectation_value)
+                else:
+                    expectation_value = np.real(
+                        wavefunction.conj().T @ sparse_pauli_term_ndarray @ wavefunction
+                    )
+                    values.append(expectation_value)
+
+            return expectation_values_to_real(ExpectationValues(np.asarray(values)))
+
+    def get_exact_noisy_expectation_values(
+        self, circuit: Circuit, qubit_operator: SymbolicOperator
+    ) -> ExpectationValues:
+        """Compute exact expectation values w.r.t. given operator in presence of noise.
+
+        Note that this method can be used only if simulator's noise_model is not set
+        to None.
+
+        Args:
+            circuit: the circuit to prepare the state
+            qubit_operator: the operator to measure
+        Returns:
+            the expectation values of each term in the operator
+        Raises:
+            RuntimeError if this simulator's noise_model is None.
+        """
+        if self.noise_model is None:
+            raise RuntimeError(
+                "Please provide noise model to get exact noisy expectation values"
+            )
+
+        cirq_circuit = cast(cirq.Circuit, export_to_cirq(circuit))
+        values = []
+
+        for pauli_term in qubit_operator:
+            sparse_pauli_term_ndarray = get_sparse_operator(
+                pauli_term, n_qubits=circuit.n_qubits
+            ).toarray()
+            if np.size(sparse_pauli_term_ndarray) == 1:
+                expectation_value = sparse_pauli_term_ndarray[0][0]
+                values.append(expectation_value)
+
+            else:
+
+                noisy_circuit = cirq_circuit.with_noise(self.noise_model)
+                rho = self._extract_density_matrix(
+                    self.simulator.simulate(noisy_circuit)
+                )
+                expectation_value = np.real(np.trace(rho @ sparse_pauli_term_ndarray))
+                values.append(expectation_value)
+        return expectation_values_to_real(ExpectationValues(np.asarray(values)))
+
+    @abc.abstractmethod
+    def _get_wavefunction_from_native_circuit(
+        self, circuit: Circuit, initial_state: StateVector
+    ) -> StateVector:
+        pass
+
+    @abc.abstractmethod
+    def _extract_density_matrix(self, result):
```

Comment:
`result` is pretty vague to me. Should that be `simulation_result` perhaps? Or something else?